### ISSUE TITLE
ORF2 not in list of livestreams

### ISSUE THREAD

**Issue by gandalth** [s0faking/plugin.video.orftvthek#135]
I just tried the plugin with this setup:

- Raspberry PI4 using Libreelec 11.0.4 and Kodi Nexus.
- Located in Austria / Austrian IP-Address
- Focus on playback of Live Channels

ORF 1, ORF III, and ORF Sport+ all work perfectly. But I'm puzzled that ORF2 is not in the folder listing the available livestreams.
Is there a known reason why ORF 2 would not be included?

Cheers,
gandalth

**Comment by s0faking** [comment]
What's the addon version? If there is no current Livestream on ORF2 it won't show up in the list

**Comment by gandalth** [comment]
The addon version is 0.12.9+matrix.1 (installed from Kodi's repository).

Reading your comment I just retried and now ORF2 is available in the list.
Still, my first test was on Saturday afternoon. ORF2 did send live broadcasting at the time. Does the live stream omit certain programming?

**Comment by s0faking** [comment]
Nope there is no filter but between shows the livestream flag is off on the Website, thats why the HTML scraper doesnt fetch the livestream parameters. I can check if i can show the channel even if there is no show running at the moment. Not sure If DRM parameters are present when the livestream Detail Pages Show a Countdown.

**Comment by gandalth** [comment]
I would be happy to test if you decide to try this.

I did more testing today, and I seem to get random playback failures, mostly on ORF1 today.
I double checked that my inputstream, inputstream-helper, and widevine installation are fine.
I now wonder if ORF's live streaming is maybe unreliable. Are there any other users with similar experience?

**Comment by s0faking** [comment]
i will look into that, but at the moment i'm trying to focus on the orf on addon which will replace this addon by april 2024. the html scraper will be left out completely on the new addon because its pretty unreliable und can't cope with small changes on the website markup very well. also it will be a lot easier for me to maintain the addon since the data source will be limited to the service api in the future, which will save me quite some headaches and mostly time ;)

can you send me a debug log for the streaming issues? there has been issues with widevine in the past, especially on the raspberry pi so this might be the problem, but i can't tell without any logs. unfortunately there is no way to fix any of this from the addon scope, since most of the streaming stuff is handled by the inputstream stack & kodi (ffmpeg) but i might be able to point out the issue if I see any obvious log output. what exactly was the behaviour, was it like stuttering video/audio stream or complete stream timeouts?

**Comment by gandalth** [comment]
I agree that your focus on the new platform makes perfect sense.
I will still paste my experience (ORF1 repeatedly says playback failed, whereas ORF2 is working). The output is the part from kodi.log where I try ORF1.

2024-01-14 18:12:52.680 T:2299  critical <general>: [script.module.inputstreamhelper] ['ldd', '/storage/.kodi/cdm/libwidevinecdm.so'] cmd failed.
2024-01-14 18:12:52.681 T:2299  critical <general>: [script.module.inputstreamhelper] Failed to check for missing Widevine libraries.
2024-01-14 18:12:52.682 T:2299  warning <general>: [script.module.inputstreamhelper] Widevine update check was made on 2024-01-14 13:11
2024-01-14 18:12:52.716 T:801      info <general>: VideoPlayer::OpenFile: plugin://plugin.video.orftvthek/?link=https%3A%2F%2Forf1.mdn.ors.at%2Forf%2Fo
rf1%2Fdrmqxb%2Fmanifest.mpd&mode=playDRM&lic_url=https%3A%2F%2Fdrm.ors.at%2Facquire-license%2Fwidevine%3FBrandGuid%3D13f2e056-53fe-4469-ba6d-999970dbe5
49%26userToken%3DollB7dLRPZujSvNM0Q%252FTIVueKtQVt%252F55CILC9N7lPZTMpV6q16yNOuUeN55jL70UuYLknGz3%252BJ3kC9SLjax7RwerFSsf8cNobNqfuzzDAFkg6L0wdrktnxWzPl
Se4%252Fc20Nw8sG%252F%252FWIXB9WJE1KzwSGOhVj6cJMci5TZexmiEORvO6u7Em2zug6ADJvC8eUuoBfMie2hqKBUDX%252FBBleI93eetEhQQ5298fN9eCxW%252B0gxGvogX8n3sNbD5GMJUB
bum
2024-01-14 18:12:52.722 T:2307     info <general>: Creating InputStream
2024-01-14 18:12:52.723 T:2307  warning <general>: AddOnLog: inputstream.adaptive: Set headers to the manifest by using "inputstream.adaptive.stream_headers" property is a deprecated behaviour that will be removed in future.
                                                   To set headers to the manifest, use "inputstream.adaptive.manifest_headers" property.
2024-01-14 18:12:52.730 T:2307     info <general>: AddOnLog: inputstream.adaptive: [Repr. chooser] Resolution set: 1920x1080, max allowed: 4096x2160, Adjust refresh rate: 0
2024-01-14 18:12:52.730 T:2307    error <general>: GetDirectory - Error getting /usr/lib/kodi/addons/inputstream.adaptive/
2024-01-14 18:12:53.477 T:2307    error <general>: CCurlFile::CReadState::FillBuffer - (0xf2d08898) Failed: HTTP returned code 403
2024-01-14 18:12:53.477 T:2307    error <general>: CCurlFile::Open - <https://orf1.mdn.ors.at/orf/orf1/drmqxb/manifest.mpd> Failed with code 403:
                                                   
2024-01-14 18:12:53.477 T:2307    error <general>: AddOnLog: inputstream.adaptive: CURLOpen returned an error, download failed: https://orf1.mdn.ors.at/orf/orf1/drmqxb/manifest.mpd
2024-01-14 18:12:53.477 T:2307    error <general>: AddOnLog: inputstream.adaptive: Could not open / parse manifest (https://orf1.mdn.ors.at/orf/orf1/drmqxb/manifest.mpd)
2024-01-14 18:12:53.478 T:2307    error <general>: CVideoPlayer::OpenInputStream - error opening [plugin://plugin.video.orftvthek/?link=https%3A%2F%2Forf1.mdn.ors.at%2Forf%2Forf1%2Fdrmqxb%2Fmanifest.mpd&mode=playDRM&lic_url=https%3A%2F%2Fdrm.ors.at%2Facquire-license%2Fwidevine%3FBrandGuid%3D13f2e056-5
3fe-4469-a6d-999970dbe549%26userToken%3DollB7dLRPZujSvNM0Q%252FTIVueKtQVt%252F55CILC9N7lPZTMpV6q16yNOuUeN55jL70UuYLknGz3%252BJ3kC9SLjax7RwerFSsf8cNobN
qfuzzDAFkg6L0wdrktnxWzPlSe4%252Fc20Nw8sG%252F%252FWIXB9WJE1KzwSGOhVj6cJMci5TZexmiEORvO6u7Em2zug6ADJvC8eUuoBfMie2hqKBUDX%252FBBleI93eetEhQQ5298fN9eCxW%2
52B0gxGvogX8n3sNbD5GMJUBbum]
2024-01-14 18:12:53.478 T:2307     info <general>: CVideoPlayer::OnExit()
2024-01-14 18:12:53.478 T:2307     info <general>: ADDON: Dll Destroyed - InputStream Adaptive
2024-01-14 18:12:53.508 T:801      info <general>: CVideoPlayer::CloseFile()
2024-01-14 18:12:53.508 T:801      info <general>: VideoPlayer: waiting for threads to exit
2024-01-14 18:12:53.508 T:801      info <general>: VideoPlayer: finished waiting

**Comment by s0faking** [comment]
hm, do you have some kind of proxy running? the 403 usually happens when fetching the playlist from a non austrian ip

if not can you turn on component specific debug logs for curl and send the log again?

**Comment by gandalth** [comment]
I just tried it with a non-austrian IP and that does not give you a 403, but redirects you to a different video. This video explains that you are not allowed to view the content outside of Austria.

I just looked at ORF TVThek through a browser (from Austria) and I think I get the problem. The current streaming of ORF1 offers coverage of a skiing event. But that only starts in a few hours. The browser shows a countdown for when the event starts. The current live program of ORF1 (a movie) is not being offered in the live streaming. I guess that is why I get the 403.
Summarizing, I had an incorrect expectation of ORF's live streaming through TVThek. I thought the whole TV program is being streamed, but actually only select shows are.
Maybe that will change with the new platform (ORF ON) you mentioned above.
